app/models/customer.rb

```diff
@@ -25,6 +25,11 @@ class Customer < ActiveRecord::Base
 
   before_create :associate_user
 
+  attr_accessor :setup_intent_client_secret
+
+  before_save :add_setup_intent, if: :allow_charges
+  before_save :remove_setup_intent, unless: :allow_charges
```

Comment:
I dont think we should add complex callbacks like this to our models and make them fat models. There must be a way to move this to a service and call it from the controller, even if just after saving the model.

Reply:
After re-reading the Strips docs, I think it's better to not store these in the database as they are intended to be short-lived objects. This also allows us to remove the callbacks and make things quite a bit simpler. It may end up that we need to store them for some reason further down the line, but for now I've just replaced it with a single `attr_accessible` and used the more generic name (`gateway_recurring_payment_client_secret`).

Reply:
ok :+1: 
Can you share a link? Setup intents are short lived? I thought they were authorizations for future recurring payments and would need to be provided later on when charging payments?

Reply:
From the [Setup Intents docs](https://stripe.com/docs/api/setup_intents): 

> Do not maintain long-lived, unconfirmed SetupIntents as they may no longer be valid. The SetupIntent then transitions through multiple statuses as it guides you through the setup process.
> Successful SetupIntents result in payment credentials that are optimized for future payments.

So the SetupIntent object is just keeping track of the process of the customer setting up the card (and verifying it in the case of SCA); once that process is complete, we would just be using the PaymentMethod itself, and the SetupIntent's `id` and `client_secret` will no longer be useful to us.